src/main/java/ch/njol/skript/lang/SkriptEvent.java

```diff
@@ -101,4 +221,34 @@ public boolean isEventPrioritySupported() {
 		return true;
 	}
 
+	/**
+	 * Fixes patterns in event by modifying every {@link ch.njol.skript.patterns.TypePatternElement}
+	 * to be nullable.
+	 */
+	public static String fixPattern(String pattern) {
+		char[] chars = pattern.toCharArray();
+		StringBuilder stringBuilder = new StringBuilder();
+
+		boolean inType = false;
+		for (int i = 0; i < chars.length; i++) {
+			char c = chars[i];
+			stringBuilder.append(c);
+
+			if (c == '%') {
+				// toggle inType
+				inType = !inType;
+
+				// add the dash character if it's not already present
+				// a type specification can have two prefix characters for modification
+				if (inType && i + 2 < chars.length && chars[i + 1] != '-' && chars[i + 2] != '-')
```

Comment:
shouldn't we only add the `-` if we're also in an optional part? like, `on world change to %string%` doesn't need to be modified, but `on world change [to %string%]` should be